### PR TITLE
EWS O365 credentials 

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Add credentials support for the integration.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EWS/ReleaseNotes/1_10_7.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### EWS O365
+- Added the *Key* integration parameter to support credentials fetching object.
```
Comment (dantavori) [inline Packs/EWS/ReleaseNotes/1_10_7.md:4]:
```suggestion
- Added the *Key / Application Secret* integration parameter to support credentials fetching object.
```

File: Packs/EWS/Integrations/EWSO365/EWSO365.py [hunk 2]
```diff
@@ -2297,8 +2306,21 @@ def sub_main():
     # client's default_target_mailbox is the authorization source for the instance
     params['default_target_mailbox'] = args.get('target_mailbox',
                                                 args.get('source_mailbox', params['default_target_mailbox']))
-    client = EWSClient(**params)
+    try:
+        client = EWSClient(**params)
```
Comment (dantavori) [inline Packs/EWS/Integrations/EWSO365/EWSO365.py:2310]: can't we add it to the try-except below?

File: Packs/EWS/Integrations/EWSO365/EWSO365_test.py [hunk 3]
```diff
@@ -360,13 +360,14 @@ def test_handle_html(mocker, html_input, expected_output):
     assert handle_html(html_input) == expected_output
 
 
-@freeze_time('2021-05-23 13:18:14.901293+00:00')
-@pytest.mark.parametrize('since_datetime, filter_arg, expected_result',
-                         [('', 'last_modified_time__gte', EWSDateTime.from_string('2021-05-23 13:08:14.901293+00:00')),
+@pytest.mark.parametrize('since_datetime, filter_arg, expected_result, parse_result',
+                         [('', 'last_modified_time__gte', EWSDateTime.from_string('2021-05-23 13:18:14.901293+00:00'),
+                           datetime.datetime(year=2021, month=0o5, day=23,
+                                             hour=13, minute=18, second=14, microsecond=901293)),
```
Comment (dantavori) [inline Packs/EWS/Integrations/EWSO365/EWSO365_test.py:366]: better to revert the change here and use freeze_time like before - please check with Itay when it's supposed to be merged and accordingly we will decide what to do

File: Packs/EWS/Integrations/EWSO365/EWSO365_test.py [hunk 4]
```diff
@@ -505,3 +510,66 @@ def test_parse_incident_from_item_with_attachments():
...
+    sub_main()
+
+    assert demisto.error.call_args[0][0] == "Exception: " + expected_result
+
+
+@pytest.mark.parametrize('params, expected_result', [
+    ({'_tenant_id': '_tenant_id', 'credentials': {'password': '1234'},
+      'default_target_mailbox': 'default_target_mailbox'}, 'ID / Application ID must be provided.')
+])
+def test_id_params(mocker, params, expected_result):
```
Comment (dantavori) [inline Packs/EWS/Integrations/EWSO365/EWSO365_test.py:540]: let's merge back the 3 tests into one (as we understood the issue is not related to them being separate, correct me if I'm wrong)

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/EWS/Integrations/EWSO365/EWSO365_test.py (hunk 4 above)
Comment (MLainer1) [reply Packs/EWS/Integrations/EWSO365/EWSO365_test.py:540]: This was a different issue that was related to the logs, so the tests still have to be separated

File: Packs/EWS/Integrations/EWSO365/EWSO365.py (hunk 2 above)
Comment (MLainer1) [reply Packs/EWS/Integrations/EWSO365/EWSO365.py:2310]: the log writing function in the second try-except interferes with the process of the exception, so you can't track the exception properly

**Comment by bakatzir** [comment]
Commenting here so I'll know when this PR is merged, as this will close the main issue and I will be able to communicate it to the eager customers.

File: Packs/EWS/Integrations/EWSO365/EWSO365.py (hunk 2 above)
Comment (MLainer1) [reply Packs/EWS/Integrations/EWSO365/EWSO365.py:2310]: I made some changes and it's now inside the second try-except

**Review by dantavori — APPROVED** [review]
we should force merge this PR since we currently can't bump the dockerimage version.

**Comment by MLainer1** [comment]
This pr needs a force-merge due to a problem in the docker image.

**Comment by Itay4** [comment]
merging due to the comment above
